node/src/main/kotlin/net/corda/node/services/persistence/DBTransactionStorage.kt

```diff
@@ -220,6 +227,17 @@ class DBTransactionStorage(private val database: CordaPersistence, cacheFactory:
         }
     }
 
+    /**
+    *  track with paging specification
+     * */
+    override fun trackWithPagingSpec(paging: PageSpecification): DataFeed<TransactionStorage.Page, SignedTransaction> {
+        return database.transaction {
+            txStorage.locked {
```

Comment:
wondering if lock is needed here, because the only place `txStorage` is used in `pagedSnapshot()` is getting the total count, unlike `snapshot()` that needs to filter through the entire map in `txStorage`.

suggest removing the lock and pass the count to the `pagedSnapshot()` method via a call param

Reply:
thanks for suggestion. made changes